httpd/handler.go

```diff
@@ -157,10 +157,18 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 // serveQuery parses an incoming query and, if valid, executes the query.
 func (h *Handler) serveQuery(w http.ResponseWriter, r *http.Request, user *influxdb.User) {
 	q := r.URL.Query()
-	p := influxql.NewParser(strings.NewReader(q.Get("q")))
-	db := q.Get("db")
+
 	pretty := q.Get("pretty") == "true"
 
+	qp := strings.TrimSpace(q.Get("q"))
```

Comment:
Because a user can send in a parameter of `q=+` which is just a space which will result in the parser still returning ok.  While not exactly the same error, the result is you really didn't give me a query parameter, so it needs to return an error.